db/migrate/20200109132519_add_sha_to_commit_deployments.rb

```diff
@@ -0,0 +1,7 @@
+class AddShaToCommitDeployments < ActiveRecord::Migration[6.0]
+  def change
+    add_column :commit_deployments, :sha, :string, limit: 40
+    add_index :commit_deployments, :sha
+    add_index :commit_deployments, [:task_id, :sha], unique: true
```

Comment:
Couple questions here:

  - Is this index (these indexes) actually needed? When do we query `commit_deployments` ?
  - Should we backfill the existing records, and then make `sha` `NOT NULL` ?

Reply:
> Is this index (these indexes) actually needed? When do we query commit_deployments ?

Good question. I’ve added these for parity with the existing association (which has equivalent indexes on `commit_id`), but will do some investigation into their use.

> Should we backfill the existing records, and then make sha NOT NULL ?

I plan this for the follow up - if I do that now, we will run the migration first and backfill the existing data, but any deployments created between `db:migrate` and the new code deploy would be created with a null sha. For safety we need to backfill the data after the code has been deployed.

Reply:
Ok I did some digging - the `commit_id` index is used only in one place that I can see, in `DestroyStackJob`. I changed that to search by task_id instead, and have removed that new index.

I'm not sure what the unique index was guarding - it doesn't look like we were catching `RecordNotUnique` anywhere in the call path, and I don't see any exceptions in Bugsnag. I've removed the new unique index, we can add it back if it causes problems in production.

Reply:
👍